src/components/Activity/ActivityHub.js

```diff
@@ -48,87 +72,109 @@ ActivityHub.propTypes = {
+  // const [showPayload, setShowPayload] = React.useState(true)
   // eslint-disable-next-line react/prop-types
-  const isDone = props => props.p === 1 && setShowPayload(false)
+  // const isDone = props => props.p === 1 && setShowPayload(false)
+
+  const app = getAppByProxyAddress(activity.targetAppProxyAddress)
 
   return (
     <Frame>
       <h1>
-        <span>{activity.description}</span>
+        <span css="flex-shrink: 0">
+          <AppIcon app={app} />
+        </span>
+        <span
+          css={`
+            margin-left: 5px;
+            white-space: nowrap;
+          `}
+        >
+          {app ? app.name : 'Unknown'}
+        </span>
       </h1>
-      <h2 />
-      <div>
-        {' '}
+      <h2>
         <TimeTag date={activity.createdAt} style={{ marginRight: 10 }} />
-        Status {activity.status}
+      </h2>
+      <div>
+        <p>{activity.description}</p>
+        <StatusMessage activity={activity} />
       </div>
-      {activity.status === activityStatusTypes.PENDING && (
-        <Spring
-          native
-          delay={400}
-          from={{ opacity: 1 }}
-          to={{ opacity: showPayload ? 1 : 0.5 }}
-        >
-          {props => (
-            <Progress style={props}>
-              <ProgressTrack>
-                <Spring
-                  native
-                  delay={ready ? 500 : 0}
-                  from={{ p: 0 }}
-                  to={{ p: ready ? 1 : 0 }}
-                  onRest={isDone}
-                >
-                  {p => (
-                    <ProgressBar
-                      style={{
-                        background: theme.accent,
-                        width: p.p.interpolate(p => `${p * 100}%`),
-                      }}
-                    />
-                  )}
-                </Spring>
-              </ProgressTrack>
-              <span>Estimated:</span>
-              <span>- min -- sec</span>
-            </Progress>
-          )}
-        </Spring>
-      )}
     </Frame>
   )
 }
 
 ActivityItem.propTypes = {
   activity: PropTypes.object.isRequired,
   ready: PropTypes.bool,
+  getAppByProxyAddress: PropTypes.func.isRequired,
+}
+
+function getStatusData(activity) {
+  const txLink = (
+    <SafeLink
+      target="_blank"
+      href={blockExplorerUrl('transaction', activity.transactionHash, {
+        networkType: network.type,
+      })}
+    >
+      transaction
+    </SafeLink>
+  )
+  if (activity.status === activityStatusTypes.CONFIRMED) {
+    return [<IconCheck />, <span>Your {txLink} is confirmed.</span>]
+  }
+  if (activity.status === activityStatusTypes.FAILED) {
+    return [<IconError />, <span>Your {txLink} failed.</span>]
+  }
+  if (activity.status === activityStatusTypes.TIMED_OUT) {
+    return [<IconError />, <span>Your {txLink} timed out.</span>]
+  }
+  return [<IconAttention />, <span>Your {txLink} is pending.</span>]
+}
+
+const StatusMessage = ({ activity }) => {
+  const [icon, content] = getStatusData(activity)
```

Comment:
It's interesting how React hooks made it a thing to return 2 value array tuples :)

Reply:
Yes totally, it’s always interesting to see patterns being upgraded from “controversial” to normal, once they reach mainstream usage and acceptation :-)